### PR TITLE
Check that code has been formatted.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
If we do not ensure that code on `main` has been formatted other users introduce unrelated changes if they do format all code. This can be avoided by running a CI check that enforce code formatting.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -742,6 +743,11 @@ format:
 	find . $(DONT_FIND) -name '*.pb.go' -prune -o -name '*.y.go' -prune -o -name '*.rl.go' -prune -o \
 		-type f -name '*.go' -exec goimports -w -local github.com/grafana/loki {} \;
 
+check-format: format
+	# Only run diff on change Go files.
+	@git diff --name-only HEAD main -- "*.go" | xargs --no-run-if-empty git diff --exit-code -- \
```
Comment (jeschkies) [inline Makefile:748]: We could use `DRONE_TARGET_BRANCH` instead of `main`.

**Review by MichelHollands — APPROVED** [review]
LGTM. Using DRONE_TARGET_BRANCH if set is probably a good idea.

File: pkg/util/marshal/output2.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...330f\U000b3628\U000de1b9\U00066c5f\U000b9093\U000dd738\U0009f13e\U000aba19䳝\U000cf135\U000e54da\U000e609d\":\"\U0004b6cb釀\U0009e058\U000b0da9\U0009c4c5\U0010fc0e𢥁\U000e5773\U000f0c3d\U000c2139\U0004636a\U0010fdc1\U000f0287𪷂𧘅\U00088202\U00071327\U0007a896閃\U00072ca2\U000eb66f\",\"\U00108854\U000b9546\U0004a437\U0005b948\U000f2f20\U000fa878\U00093eba\U000b1baf\U0001b7b3\U000150a6\U0001c7ca\U000a9b41\U0004dfcc\U000f66df\U00100443\U00080675\U000babf0𝧸\U000827e9\U00046893\U000f2341\U0007bf0b\U000f2b86\U000ccd7d\U0003e66d\":\"𖡁\U0001de17\U000cbdfe\U0006d5a1뤴\U0002fe6c\U0007bd5f\U0009c683\U000cfe83\U000c6ea6\U00039b4c\U000f5393\U000c2b73\U000e076d𘦲\U00016615\U00011bbf\U00083587\U000c282d\U0009949b\U000bb5c4\U00079d70\U00044b64\U000deddd\U000e4f4b\U0008a60c\U00015c72\U000ff9fd\U0010ed02𧷈\U0010e9a9\U0005666d\U0001c741\U000a50fc𡇪\U000f5ab8\U00055fc1\"},\"value\":[8700018213312418,\"-64473510413719020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"]}]"
```
Comment (jeschkies) [inline pkg/util/marshal/output2.txt:1]: That's some rebasing maddness. Will remove once the check worked.

**Comment by jeschkies** [comment]
It worked https://drone.grafana.net/grafana/loki/22226/3/14 :tada:

**Comment by jeschkies** [comment]
> Good on ya for adding this!
> 
> 
> 
> If we're going to add this, we should ensure that all code is already formatted so we're starting from a good base. Could we do that as a precursor to this PR?

That was my original idea. However, it would mess up git blame. So the idea is that only files changed in this commit are checked for being formatted. That's why there's this odd git diff command. I think ./pkg is pretty far. ./clients and ./operator are lagging behind

**Pull request merged**